arch/Kconfig

```diff
@@ -104,6 +104,7 @@ config RISCV
 config XTENSA
 	bool
 	select ARCH_IS_SET
+	select ARCH_SUPPORTS_COREDUMP if SOC_XTENSA_SAMPLE_CONTROLLER || SOC_ESP32
```

Comment:
we should not make this conditional on SOC, this list is going to grow and that will become unmanageable. Instead I would put this in the documentation, ie. what boards currently work.

Reply:
Or, do the `select` in the SoC layer instead.

Reply:
BTW, don't add this line in this commit until the support is actually there.

Reply:
I combined the Kconfig changes with the PR that introduced the coredump (but not the GDB server) and added a Kconfig option `SOC_SUPPORTS_COREDUMP` selected by `SOC_ESP32` and `SOC_XTENSA_SAMPLE_CONTROLLER` and depended on by `DEBUG_COREDUMP`, let  me know what you think.